src/pybind/mgr/cephadm/module.py

```diff
@@ -2873,12 +2873,44 @@ def apply_drivegroups(self, specs: List[DriveGroupSpec]) -> List[str]:
         """
         return [self._apply(spec) for spec in specs]
 
+    def validate_device(self, host_name: str, drive_group: DriveGroupSpec) -> str:
+        try:
+
+            if not drive_group.data_devices or not drive_group.data_devices.paths:
+                return "Error: No data devices specified."
+
+            if self.cache.is_host_unreachable(host_name):
+                return f"Host {host_name} is not reachable (it may be offline or in maintenance mode)."
+
+            host_cache = self.cache.devices.get(host_name, [])
+            if not host_cache:
+                return f"Error: No devices found for host {host_name}."
+
+            available_devices = {
+                dev.path: dev for dev in host_cache if dev.available
+            }
+            self.log.debug(f"Available devices: {host_name}: {list(available_devices)}")
```

Comment:
this log could be very long in case we have a lot of devices! ... not sure it's useful TBH in case we already have some command to list them.